### PR TITLE
Prevent content stretching on fronts when ad-free removes MPUs

### PR DESCRIPTION
**Pull request by JustinPinner**
## What does this change?
Switches content layouts if MPUs are removed - following the existing `commercialOptions.omitMPU` pattern.

## What is the value of this and can you measure success?
Prevents content stretching to fill space vacated by removed MPUs

## Does this affect other platforms - Amp, Apps, etc?
No, web-only.

## Screenshots
**Signed-out user (default behaviour)**
screen shot 2017-07-25 at 11 02 11

**Signed-in user (ad-free) - BEFORE**
screen shot 2017-07-25 at 11 47 30

**Signed-in user (ad-free) - AFTER**
screen shot 2017-07-25 at 11 02 29

## Tested in CODE?
Yes

### CODE REVIEW

File: common/app/views/fragments/containers/index.scala.html [hunk 1]
```diff
@@ -11,7 +11,7 @@
 @* TODO Consolidate this with the master container template to get rid of repetition *@
 
 @defining(page.webTitle) { case (webTitle) =>
-    <section class="@GetClasses.forTagContainer(webTitle.nonEmpty)"
+    <section class="@GetClasses.forTagContainer(webTitle.nonEmpty, adFree = request.headers.keys.exists(_ equalsIgnoreCase "X-Gu-Commercial-Ad-Free"))"
```
Comment (TBonnin) [inline common/app/views/fragments/containers/index.scala.html]: would it make sense to an `adFree` value in https://github.com/guardian/frontend/blob/513e751f93f74db344cc7faf66f0c4227a4adf34/common/app/implicits/Requests.scala

Comment (JustinPinner) [inline common/app/views/fragments/containers/index.scala.html]: Quite probably. I'm not sure the request implicit is passed deep enough into the backend code to be universally available - there were places it didn't seem to be referenced, but at this point, yes it could definitely read a property from there.

**Comment by paperboyo** [comment]
💎! Does it mean that the space will be filled with content proper? Will it work like that in non-ad-free-but-with-adblock scenario, too?

So happy to see less elongated cards and low-res images! 😄

File: common/app/views/fragments/containers/index.scala.html (hunk 1 above)
Comment (JustinPinner) [inline common/app/views/fragments/containers/index.scala.html]: Actually, at this point, `request` seems to be a plain `play.api.mvc.RequestHeader` so the properties of `RichRequestHeader` are not available. However, it did work for https://github.com/guardian/frontend/pull/17502/files#diff-9e483d0f7f6313cb58e2a1515b4ee26fR194

Comment (TBonnin) [inline common/app/views/fragments/containers/index.scala.html]: you can import `Requests._` wherever you need it

Comment (JustinPinner) [inline common/app/views/fragments/containers/index.scala.html]: But there are many places where doing so would require changing a ton of downstream classes etc, that have no need to know about the new parameter (or any of the others).

**Comment by JustinPinner** [comment]
@paperboyo YES :) Or that's the plan anyway. I'm just deploying to CODE again now if you want to have a 👀

File: common/app/views/fragments/containers/index.scala.html (hunk 1 above)
Comment (JustinPinner) [inline common/app/views/fragments/containers/index.scala.html]: By the way @TBonnin I'm not suggesting this is a bad thing to do, but maybe if we do it it should be as a separate PR? I think having a consistent request header construct everywhere would be good in the long run, no?

Comment (TBonnin) [inline common/app/views/fragments/containers/index.scala.html]: I don't really understand: You just need to `import implicits.Requests._` and you can then call `request.isAdFree` where needed

Comment (JustinPinner) [inline common/app/views/fragments/containers/index.scala.html]: Maybe I was doing it wrong. I'll have another go.

File: sport/app/football/containers/FixturesAndResults.scala [hunk 2]
```diff
@@ -41,6 +42,8 @@ class FixturesAndResults(competitions: Competitions) extends Football {
 
   def makeContainer(tagId: String)(implicit request: RequestHeader, context: ApplicationContext) = {
 
+    lazy val adFree = request.isAdFree
```
Comment (TBonnin) [inline sport/app/football/containers/FixturesAndResults.scala]: does this lazy val necessary?

Comment (JustinPinner) [inline sport/app/football/containers/FixturesAndResults.scala]: Nope. Consider it gone! (Thanks)

**Review by TBonnin — APPROVED** [review]

**Review by kelvin-chappell — APPROVED** [review]
I wonder whether the omitMPU flag could have been set either when there was a takeover or when it was an ad-free request?  So for any scenario where MPUs are to be removed.  Then there would be no need to add the extra adFree property further downstream.  But apart from that it looks good so 👍

**Comment by JustinPinner** [comment]
Thanks @kelvin-chappell. As discussed, I had first thought about wrapping the ad-free requirements up with omitMPU but then if someone in the future wanted to change one they must also be aware of the other. Admittedly this adds more code but keeps the two concepts isolated all the way down.
